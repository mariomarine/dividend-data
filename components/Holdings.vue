<template>
	<div class="holdings">
		<h2>Holdings</h2>
		<div class="holdings-table">
			<div class="holdings-row">
				<span> Ticker </span>
				<span> Shares </span>
				<span> Price </span>
			</div>
			<Holding>
				<span slot="ticker"> {{ ticker }} </span>
				<span slot="shares"> {{ shares }} </span>
				<span slot="price"> $84.87 </span>
			</Holding>
			<Holding>
				<span slot="ticker"> DUK </span>
				<span slot="shares"> 14 </span>
				<span slot="price"> $114.91 </span>
			</Holding>
			<Holding>
				<span slot="ticker"> NEE </span>
				<span slot="shares"> 15 </span>
				<span slot="price"> $11.91 </span>
			</Holding>
		</div>
	</div>
</template>

<script>
import Holding from '@/components/Holding.vue'

export default {
	components: {
		Holding
	},
	data() {
		return {
			ticker: 'SO',
			shares: 28.1
		}
	}
}
</script>

<style>
.holdings-table {
	display: grid;
	grid-template-rows: auto;
}
.holdings-row {
	display: grid;
	grid-template-columns: 1fr 1fr 1fr;
}
</style>

