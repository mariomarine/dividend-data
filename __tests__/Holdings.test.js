import { mount } from '@vue/test-utils';
import Holdings from '../components/Holdings.vue';

describe('Holdings.vue', () => {
  test('should mount for testing', () => {
    expect(1).toEqual(1);
  });
  test('should render content correctly', () => {
	const wrapper = mount(Holdings);
	expect(wrapper.vm.$el).toMatchSnapshot();
  });
});

