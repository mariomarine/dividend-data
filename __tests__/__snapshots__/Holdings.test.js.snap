// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Holdings.vue should render content correctly 1`] = `
<div
  class="holdings"
>
  <h2>
    Holdings
  </h2>
   
  <div
    class="holdings-table"
  >
    <div
      class="holdings-row"
    >
      <span>
         Ticker 
      </span>
       
      <span>
         Shares 
      </span>
       
      <span>
         Price 
      </span>
    </div>
     
    <div
      class="holdings-row"
    >
      <span>
         SO 
      </span>
       
      <span>
         28.1 
      </span>
       
      <span>
         $84.87 
      </span>
    </div>
     
    <div
      class="holdings-row"
    >
      <span>
         DUK 
      </span>
       
      <span>
         14 
      </span>
       
      <span>
         $114.91 
      </span>
    </div>
     
    <div
      class="holdings-row"
    >
      <span>
         NEE 
      </span>
       
      <span>
         15 
      </span>
       
      <span>
         $11.91 
      </span>
    </div>
  </div>
</div>
`;
